decidim-proposals/app/assets/javascripts/decidim/proposals/admin/proposals.js.erb

```diff
@@ -0,0 +1,112 @@
+// = require_self
+
+let selectedProposalsCount = function() {
+  return $('.table-list .js-check-all-proposal:checked').length
+}
+
+let selectedProposalsCountUpdate = function() {
+  if(selectedProposalsCount() == 0){
+    $("#js-recategorize-proposals-count").text("")
+  } else {
+    $("#js-recategorize-proposals-count").text(selectedProposalsCount());
+  }
+}
+
+let showBulkActionsButton = function() {
+  if(selectedProposalsCount() > 0){
+    $("#js-bulk-actions-button").removeClass('hide');
+  }
+}
+
+let hideBulkActionsButton = function(force = false) {
+  if(selectedProposalsCount() == 0 || force == true){
+    $("#js-bulk-actions-button").addClass('hide');
+    $("#js-bulk-actions-dropdown").removeClass('is-open');
+  }
+}
+
+let showOtherActionsButtons = function() {
+  $("#js-other-actions-wrapper").removeClass('hide');
+}
+
+let hideOtherActionsButtons = function() {
+  $("#js-other-actions-wrapper").addClass('hide');
+}
+
+let showRecategorizeProposalActions = function() {
+  $("#js-recategorize-proposals-actions").removeClass('hide');
+}
+
+let hideRecategorizeProposalActions = function() {
+  $("#js-recategorize-proposals-actions").addClass('hide');
+}
+
+
+if ($('#js-form-recategorize-proposals').length) {
+  hideRecategorizeProposalActions();
+  $("#js-bulk-actions-button").addClass('hide');
+
+  $("#js-bulk-actions-recategorize").click(function(e){
+    e.preventDefault();
+
+    $('#js-form-recategorize-proposals').submit(function(){
+      $('.layout-content > .callout-wrapper').html("");
+    })
+
+    showRecategorizeProposalActions();
+    hideBulkActionsButton(true);
+    hideOtherActionsButtons();
+  })
+
+  // select all checkboxes
+  $(".js-check-all").change(function() {
```

Comment:
Shouldn't this delegate instead (use `on`)? Otherwise it won't work if the element doesn't exist.

Reply:
the `.js-check-all` is (or should) always be present as it's the `checkbox` rendered in the `<thead>` that once clicked clicks the rest on the `tbody > tr` that can be reloaded by `ajax`.
`js` is only responsible for (un)checking the `.js-check-all` `checkbox`.

I think there is no need to delegate it, but if you consider I'll add it